generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -22,62 +28,56 @@
         null,
         FieldSpecSource.Empty);
 
-    private final SetRestrictions setRestrictions;
-    private final NumericRestrictions numericRestrictions;
-    private final StringRestrictions stringRestrictions;
-    private final NullRestrictions nullRestrictions;
-    private final TypeRestrictions typeRestrictions;
-    private final DateTimeRestrictions dateTimeRestrictions;
-    private final FormatRestrictions formatRestrictions;
-    private final MustContainRestriction mustContainRestriction;
-    private final FieldSpecSource source;
+    @Nullable private final SetRestrictions setRestrictions;
```

Comment:
what's the point of @ Nullable

Reply:
Disambiguates between "this can be null, and that's meaningful" and "this can be a null, and oh I didn't think about that".

Reply:
For who the reader or the IDE?

Reply:
Is there any benefit using @ Nullable, over using Optional?

Reply:
Optional-typed fields can still be null :sweat_smile: Also I think there's a recommendation against using optionals for fields - it's more intended for return types? Not sure I follow the reasoning behind that, though.

Reply:
(more generally: if we agreed that re-engineering to use Optional were the right approach, would we fail this PR? it's dangerous to get into a mentality of "small improvements are rejected if there's a larger improvement you could make instead")

Reply:
I wouldn't reject this PR based on that. There are several ways to achieve the desired outcome, and I was just curious why you went one way instead of another.

Reply:
Makes sense - I was just trying to minimise the scope of the change. I think there's definite improvement to be made to this entire class (it's weird that one class represents two things - whitelist and blacklist - in a completely exclusive way; it's begging to be broken into two classes, but that involves more changes elsewhere).